src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11950,6 +11976,11 @@ private bool IsPossibleLambdaExpression(Precedence precedence)
                 var attributeDeclarations = ParseAttributeDeclarations(inExpressionContext: true);
                 if (attributeDeclarations is [.., { CloseBracketToken.IsMissing: true }])
                     return false;
+
+                // If we didn't even end up with any attributes, then this definitely is not an attribute decl on a
+                // lambda. This is just a collection expr.
+                if (attributeDeclarations.Count == 0)
+                    return false;
```

Comment:
> return false; [](http://example.com/codeflow?start=19&length=14)

Do we have a test covering this code path?

Reply:
No.  An affectation from a previous path.  I've removed it.  Thanks!